Arduino Uno R3/examples/ADE9153A_examples/ADE9153AAPI_Test/ADE9153AAPI_Test.ino

```diff
@@ -0,0 +1,202 @@
+#define CS_PIN 8              //8-->Arduino Zero. 15-->ESP8266 
+#define ADE9153A_RESET_PIN 4  //On-board Reset Pin
+#define USER_INPUT 5          //On-board User Input Button Pin
+#define LED 6                 //On-board LED pin
+
+EnergyRegs energyVals;  //Energy register values are read and stored in EnergyRegs structure
+PowerRegs powerVals;    //Metrology data can be accessed from these structures
+RMSRegs rmsVals;  
+PQRegs pqVals;
+AcalRegs acalVals;
+Temperature tempVal;
+
+void readandwrite(void);
+void resetADE9153A(void);
+
+int ledState = LOW;
+int inputState = LOW;
+unsigned long lastReport = 0;
+const long reportInterval = 2000;
+const long blinkInterval = 500;
+
+void setup() {
+  /*Pin and serial monitor setup*/
+  pinMode(LED, OUTPUT);
+  pinMode(USER_INPUT, INPUT);
+  pinMode(ADE9153A_RESET_PIN, OUTPUT);
+  digitalWrite(ADE9153A_RESET_PIN, HIGH);  
+  Serial.begin(115200);
+
+  resetADE9153A();            //Reset ADE9153A for clean startup
+  delay(1000);
+  /*SPI initialization and test*/
+  bool commscheck = ade9153A.SPI_Init(SPI_SPEED,CS_PIN); //Initialize SPI
+  if (!commscheck) {
+    Serial.println("ADE9153A Shield not detected. Plug in Shield and reset the Arduino");
+    while (!commscheck) {     //Hold until arduino is reset
+      delay(1000);
+    }
+  }
+  
+  ade9153A.SetupADE9153A(); //Setup ADE9153A according to ADE9153AAPI.h
+  delay(500); 
+}
+
+void loop() {
+  /*Main loop*/
+  /*Returns metrology to the serial monitor and waits for USER_INPUT button press to run autocal*/
+  unsigned long currentReport = millis();
+  
+  if ((currentReport - lastReport) >= reportInterval){
+    lastReport = currentReport;
+    readandwrite();
+  }
+  
+  inputState = digitalRead(USER_INPUT);
+
+  if (inputState == LOW) {
+    Serial.println("Autocalibrating Current Channel");
+    ade9153A.StartAcal_AINormal();
+    runLength(20);
+    ade9153A.StopAcal();
+    Serial.println("Autocalibrating Voltage Channel");
+    ade9153A.StartAcal_AV();
+    runLength(40);
+    ade9153A.StopAcal();
+    delay(100);
+    
+    ade9153A.ReadAcalRegs(&acalVals);
+    Serial.print("AICC: ");
+    Serial.println(acalVals.AICC);
+    Serial.print("AICERT: ");
+    Serial.println(acalVals.AcalAICERTReg);
+    Serial.print("AVCC: ");
+    Serial.println(acalVals.AVCC);
+    Serial.print("AVCERT: ");
+    Serial.println(acalVals.AcalAVCERTReg);
+    long Igain = (-(acalVals.AICC / 838.190) - 1) * 134217728;
+    long Vgain = ((acalVals.AVCC / 13411.05) - 1) * 134217728;
+    ade9153A.SPI_Write_32(REG_AIGAIN, Igain);
+    ade9153A.SPI_Write_32(REG_AVGAIN, Vgain);
+    //ade9153A.ApplyAcal(acalVals.AICC, acalVals.AVCC);  //The ApplyAcal command does not account for a reversed IAN-IAP connection
+    
+    Serial.println("Autocalibration Complete");
+    delay(2000);
+  }
+}
+
+void readandwrite()
+{
+
+ /*Read and Print Specific Register using ADE9153A SPI Library */
+  //Serial.println(ade9153A.SPI_Read_32(REG_AIRMS),HEX); // AIRMS
```

Comment:
Since this is example code for using the library I wanted to show what an individual register read looks like, maybe the better way to handle this would be to put a line in setup that reads the product version register and sends it across the serial port.

Reply:
that would work.